### PR TITLE
Storybook: Add stories for the TextDecorationControl component

### PR DESCRIPTION
**Pull request by himanshupathak95**
Part of https://github.com/WordPress/gutenberg/issues/67165

## What?
This PR improves the Storybook stories for the TextDecorationControl component in the block editor. It includes stories demonstrating different text-decoration states.

## Why?
As part of the ongoing effort to improve component documentation and testing (tracked in #22891), we need comprehensive stories for all Block Editor components.

## Testing Instructions
- Start Storybook by running `npm run storybook:dev`
- Open Storybook at http://localhost:50240/
- Navigate to "BlockEditor/TextDecorationControl" in the Storybook sidebar
- Test the following stories:
    - Default (no decoration)
    - WithUnderline (underline decoration)
    - WithStrikethrough (strikethrough decoration)
    - WithCustomClass (custom class name)

## Screencast 

https://github.com/user-attachments/assets/96a5f9a0-d615-45ba-aa5b-419d1c46d6ec

### CODE REVIEW

**Review by t-hamano — COMMENTED** [review]
Thanks for the PR! 

I would like to make two suggestions.

- It looks like you are adding as many variations as possible as individual stories, but this seems redundant. All these props can be changed from the Controls panel. Also, if we go ahead with this approach, we will probably add similar variations to all other block editor components 😅
- It is certainly easier to understand with demo text, but Storybook is for testing and developing a single component itself. How this component interacts with the outside world is the responsibility of the developer who uses this component, and I think it is better not to include it in the story.

With that in mind, I think it would be best to focus on just the following two improvements for this PR. What do you think?

- Add the missing propsI (`value`, `className`) to `argTypes`
- Convert to CSF3 format (See [this comment](https://github.com/WordPress/gutenberg/issues/67165#issuecomment-2511062105))

**Comment by himanshupathak95** [comment]
Thanks, @t-hamano for the suggestions.

I have addressed all the comments. Please let me if we can improve further.

https://github.com/user-attachments/assets/bc34a7b3-dcdf-46d0-81a1-478110311b27

File: packages/block-editor/src/components/text-decoration-control/stories/index.story.js [hunk 1]
```diff
@@ -8,26 +8,65 @@ import { useState } from '@wordpress/element';
...
 	argTypes: {
-		onChange: { action: 'onChange' },
+		value: {
+			control: { type: null },
+			description: 'Currently selected text decoration.',
+			table: {
+				type: {
+					summary: 'string',
+				},
+				defaultValue: { summary: 'none' },
```
Comment (t-hamano) [inline packages/block-editor/src/components/text-decoration-control/stories/index.story.js]:
```suggestion
				},
```

I think this is a mistake in the README. [The default value is undefined](https://github.com/WordPress/gutenberg/blob/2ea929cef1d225d4a075269453c5f570b874dddb/packages/block-editor/src/components/text-decoration-control/index.js#L45), not the string `"none"`.

Besides this part, could you also remove this line below?

https://github.com/WordPress/gutenberg/blame/widgets/fix/inserter-open/packages/block-editor/src/components/text-decoration-control/README.md#L31

File: packages/block-editor/src/components/text-decoration-control/stories/index.story.js [hunk 2]
```diff
@@ -8,26 +8,65 @@ import { useState } from '@wordpress/element';
...
-		/>
-	);
-};
+export default meta;
 
-export const Default = Template.bind( {} );
+/**
+ * Default story showing TextDecorationControl
+ */
+export const Default = {
```
Comment (t-hamano) [inline packages/block-editor/src/components/text-decoration-control/stories/index.story.js]:
```suggestion
export const Default = {
```

There is only one story, so this comment is no longer necessary.

File: packages/block-editor/src/components/text-decoration-control/stories/index.story.js [hunk 3]
```diff
@@ -8,26 +8,65 @@ import { useState } from '@wordpress/element';
...
+	render: function Template( { onChange, ...args } ) {
+		const [ value, setValue ] = useState( 'none' );
```
Comment (t-hamano) [inline packages/block-editor/src/components/text-decoration-control/stories/index.story.js]:
```suggestion
		const [ value, setValue ] = useState();
```

File: packages/block-editor/src/components/text-decoration-control/stories/index.story.js [hunk 4]
```diff
@@ -8,26 +8,65 @@ import { useState } from '@wordpress/element';
...
+		return (
+			<TextDecorationControl
+				{ ...args }
+				onChange={ ( newValue ) => {
+					onChange( newValue );
+					setValue( newValue );
+				} }
```
Comment (t-hamano) [inline packages/block-editor/src/components/text-decoration-control/stories/index.story.js]:
```suggestion
				onChange={ ( ...changeArgs ) => {
					onChange( ...changeArgs );
					setValue( ...changeArgs );
				} }
```

This change isn't required, but let's make it consistent with the other stories.

**Review by t-hamano — COMMENTED** [review]
Thanks for the PR! It looks almost good, but I'd like to give you some additional feedback 🙏

File: packages/block-editor/src/components/text-decoration-control/stories/index.story.js (hunk 1 above)
Comment (himanshupathak95) [reply packages/block-editor/src/components/text-decoration-control/stories/index.story.js]: Sure, @t-hamano I have made changes to the README.md as well.

**Comment by himanshupathak95** [comment]
I have addressed all the comments. Thanks a lot, @t-hamano for the additional suggestion and for improving the overall structure of the story 🙌🏻

**Review by t-hamano — COMMENTED** [review]
LGTM 👍

**Review by t-hamano — APPROVED** [review]

**Pull request merged**